crypto/dh/dh_gen.c

```diff
@@ -88,7 +88,7 @@ static int dh_builtin_genparams(DH *ret, int prime_len, int generator,
             goto err;
         g = 2;
     } else if (generator == DH_GENERATOR_5) {
-        if (!BN_set_word(t1, 10))
+        if (!BN_set_word(t1, 20))
```

Comment:
Why do you change add from 10 to 20?

Reply:
As said in the commit message, there are no safe primes == 13 mod 20,
but with since probable_prime_dh (safe) does not (did never) check that q is divisible by 2
there are a lot of candidates p=1 mod 4, and the first round of miller rabin takes a lot
of time, since it checks p, while the check for q fails immediately because it is even.

Reply:
Can update the comment at the top of the function?

Reply:
Do you have any reference that they are not a safe prime?

Reply:
Never mind

Reply:
> Can update the comment at the top of the function?

Yes.

I wonder if I should change the (unreachable) default case below
from 1 mod 2 to 11 mod 12?

Reply:
I think we actually want to check that == 23 mod 60 (or 47 mod 60), since 3, 13, 33, 43 and 53 mod 60 are not safe primes.

Reply:
Yes, we want 11 mod 12 in the default case.